### PR TITLE
[PWC-54] Category feature

### PR DESCRIPTION
**Pull request by dyudyunov**
Course categories, Categories list page, Category page added, general
fixes applied.

**Youtrack:** https://youtrack.raccoongang.com/issue/FPWC-53

**Author concerns:** For correct work of navbar -> go to admin page > programs api config > “Do we want to show program listing page:” delete flag

Course categories was not deleted from the main page

### CODE REVIEW

File: common/djangoapps/course_category/views.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        visible_courses = get_courses(request.user)
+        categories = CourseCategory.objects.filter(parent=None, courses__in=visible_courses).distinct()
+        context = {'categories': categories}
+        return render_to_response("course_category_list.html", context)
+
+
+class CategoryCoursesListView(View):
+
+    def get(self, request, slug, *args, **kwargs):
+
```
Comment (drtwisted) [inline common/djangoapps/course_category/views.py]:
```suggestion
```

File: common/djangoapps/course_category/tasks.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from celery.task import task
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+
+
+@task
+def task_reindex_courses(category_id=None, course_keys=None):
+    from course_category.models import CourseCategory
+    from cms.djangoapps.contentstore.courseware_index import CoursewareSearchIndexer
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply common/djangoapps/course_category/tasks.py]: I saw imports inside task functions before in the project, but I have no 100% answer.

File: cms/djangoapps/contentstore/courseware_index.py [hunk 3]
```diff
@@ -591,7 +594,8 @@ def index_about_information(cls, modulestore, course):
             'course': course_id,
             'content': {},
             'image_url': course_image_url(course),
-            'catalog_visibility': course.catalog_visibility
+            'catalog_visibility': course.catalog_visibility,
+            'category': []
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply cms/djangoapps/contentstore/courseware_index.py]: Was discussed and decided to leave like it is.

File: common/djangoapps/course_category/tasks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dyudyunov) [reply common/djangoapps/course_category/tasks.py]: @UvgenGen said that it causes cyclic import error if move it to the top.

[the remark this answers is not in the document]
Comment (cmltaWt0) [reply common/djangoapps/course_category/tasks.py]: Yeap. I know this is done to avoid cyclic imports.

**Pull request merged**